scripts/interfaces.ts

```diff
@@ -0,0 +1,167 @@
+    "chains/arbitrum/FusePoolDirectoryArbitrum.sol",
+    "core/ComptrollerStorage.sol",
+    "core/CTokenInterfaces.sol",
+    "core/ErrorReporter.sol",
+    "core/RewardsDistributorStorage.sol",
+    "src/FusePoolLens.sol",
+    "src/FusePoolLensSecondary.sol",
+    "src/FuseSafeLiquidator.sol",
+  ];
+
+  const PROJECT_ROOT_DIR = `${__dirname}/..`;
+  const ABI_DIR = `${PROJECT_ROOT_DIR}/abi`;
+  const INTERFACES_DIR = `${PROJECT_ROOT_DIR}/src/test/interfaces`;
+  const SCRIPTS_DIR = `${PROJECT_ROOT_DIR}/scripts`;
+
+  const FILEPATHS = glob
+    .sync(`${PROJECT_ROOT_DIR}/src/**/*.sol`, {
+      nodir: true,
+    })
+    .filter(
+      (filePath) =>
+        !IGNORE_LIST.map((item) => filePath.includes(item)).includes(true)
+    );
+
+  const PREVIOUS_HASHES = JSON.parse(
+    await readFile(`${SCRIPTS_DIR}/hashes.json`, "utf-8")
+  );
+
+  const NEW_HASHES = Object.assign(
+    {},
+    ...(await Promise.all(
+      FILEPATHS.map((filePath) => createHashFromFile(filePath))
+    ))
+  );
+
+  const DIFF_HASHES = Object.entries({
+    ...PREVIOUS_HASHES,
+    ...NEW_HASHES,
+  }).reduce((acc, [key, value]) => {
+    if (
+      !Object.values(PREVIOUS_HASHES).includes(value) ||
+      !Object.values(NEW_HASHES).includes(value)
+    )
+      (acc as any)[key] = value;
+
+    return acc;
+  }, {});
+
+  // Filter filepaths down to diff hashes
+  const DIFF_HASHES_PATHS = Object.keys(DIFF_HASHES)
+    .map((item) =>
+      FILEPATHS.filter((filePath) => filePath.includes(`/${item}`))
+    )
+    .flat();
+
+  if (DIFF_HASHES_PATHS.length === 0) {
+    logger.info("No changes found, exiting...");
+    return;
+  } else {
+    logger.info(`Re-creating ABI and Interfaces for [${DIFF_HASHES_PATHS}]`);
+  }
+
+  for (const filePath of DIFF_HASHES_PATHS) {
+    if (IGNORE_LIST.map((item) => filePath.includes(item)).includes(true)) {
+      continue;
+    }
+
+    const fileName = parse(basename(filePath)).name;
+
+    const abiOutputPath = `${ABI_DIR}/${filePath
+      .split("/src/")
+      .pop()
+      ?.replace(".sol", ".json")}`;
+
+    const interfaceOutputPath = `${INTERFACES_DIR}/${filePath
+      .split("/src/")
+      .pop()
+      ?.replace(fileName, `I${fileName}`)}`;
+
+    const dirPath = dirname(filePath.split("/src/")[1]);
+
+    if (!abiOutputPath || !interfaceOutputPath) {
+      continue;
+    }
+
+    try {
+      const abiOutput = (await spawnProcess(
+        `forge inspect ${fileName} abi`
+      ).catch((error) => {
+        logger.warn(error);
+        return "";
+      })) as string;
+
+      if (!abiOutput) {
+        logger.warn(`Failed to generate ABI for ${filePath}, skipping...`);
+        continue;
+      }
+
+      // Write ABI
+      mkdirSync(`${ABI_DIR}/${dirPath}`, { recursive: true });
+
+      await writeFile(abiOutputPath, abiOutput);
+
+      const rawInterfaceOutput = (await spawnProcess(
+        `cast interface ${abiOutputPath}`
+      ).catch((error) => {
+        logger.warn(error);
+        return "";
```

Comment:
Not ideal to return and empty string but allows for easy checking and avoids more type checks